setup-env

```diff
@@ -0,0 +1,37 @@
+# This script sets up an environment suitable for development.
+#
+# Usage:
+#
+#   $ source setup-env
+#
+
+_xw_die() {
+    echo "Failed to set up environment!" >&2
+    return
+}
+
+_XW_REPO_ROOT="$(git rev-parse --show-toplevel)"
+_XW_VENV_ROOT="$_XW_REPO_ROOT/venv/crosswords"
+_XW_PYTHON=python3.10
+
+# Create a virtual environment if we don't already have one ...
+if [[ ! -e "$_XW_VENV_ROOT/bin/activate" ]]; then
+    echo "Creating a virtual environment at '$_XW_VENV_ROOT'" >&2
+    rm -rf "$_XW_VENV_ROOT" || _xw_die
+    "$_XW_PYTHON" -m venv "$_XW_VENV_ROOT" || _xw_die
+fi
+
+# ... activate it ...
+source "$_XW_VENV_ROOT/bin/activate" || _xw_die
```

Comment:
Not sure if you've really worked with virtual environments before (here's the [stdlib docs](https://docs.python.org/3/tutorial/venv.html)), but after this line, we can just refer to `pip` and `python`, and we won't conflict with any "system-wide" Python packages